xdem/coreg/biascorr.py

```diff
@@ -317,9 +322,114 @@ def _fit_func(  # type: ignore
+        ref_elev: NDArrayf,
+        tba_elev: NDArrayf,
+        inlier_mask: NDArrayb,
+        transform: rio.transform.Affine,
+        crs: rio.crs.CRS,
+        z_name: str,
+        weights: NDArrayf | None = None,
+        bias_vars: dict[str, NDArrayf] | None = None,
+        verbose: bool = False,
+        **kwargs: Any,
+    ) -> None:
+        """Should only be called through subclassing"""
+
+        self._fit_biascorr(
+            ref_elev=ref_elev,
+            tba_elev=tba_elev,
+            inlier_mask=inlier_mask,
+            transform=transform,
+            crs=crs,
+            z_name=z_name,
+            weights=weights,
+            bias_vars=bias_vars,
+            verbose=verbose,
+            **kwargs,
+        )
+
+    def _fit_rst_pts(  # type: ignore
+        self,
+        ref_elev: NDArrayf | gpd.GeoDataFrame,
+        tba_elev: NDArrayf | gpd.GeoDataFrame,
+        inlier_mask: NDArrayb,
+        transform: rio.transform.Affine,  # Never None thanks to Coreg.fit() pre-process
+        crs: rio.crs.CRS,  # Never None thanks to Coreg.fit() pre-process
+        z_name: str,
+        bias_vars: None | dict[str, NDArrayf] = None,
+        weights: None | NDArrayf = None,
+        verbose: bool = False,
+        **kwargs,
+    ) -> None:
+        """Should only be called through subclassing."""
+
+        # Get point reference to also convert inlier and bias vars
+        pts_elev = ref_elev if isinstance(ref_elev, gpd.GeoDataFrame) else tba_elev
+        rst_elev = ref_elev if not isinstance(ref_elev, gpd.GeoDataFrame) else tba_elev
+
+        pts = np.array((pts_elev.geometry.x.values, pts_elev.geometry.y.values)).T
+
+        # Get valid mask ahead of subsampling to have the exact number of requested subsamples by user
+        if bias_vars is not None:
+            valid_mask = np.logical_and.reduce(
+                (inlier_mask, np.isfinite(rst_elev), *(np.isfinite(var) for var in bias_vars.values()))
+            )
+        else:
+            valid_mask = np.logical_and.reduce((inlier_mask, np.isfinite(rst_elev)))
+
+        # Convert inlier mask to points to be able to determine subsample later
+        inlier_rst = gu.Raster.from_array(data=valid_mask, transform=transform, crs=crs)
+        # The location needs to be surrounded by inliers, use floor to get 0 for at least one outlier
+        valid_pts = np.floor(inlier_rst.interp_points(pts)).astype(bool)  # Interpolates boolean mask as integers
+
+        # If there is a subsample, it needs to be done now on the point dataset to reduce later calculations
+        subsample_mask = self._get_subsample_on_valid_mask(valid_mask=valid_pts, verbose=verbose)
+        pts = pts[subsample_mask]
+
+        # Now all points should be valid, we can pass an inlier mask completely true
+        inlier_pts_alltrue = np.ones(len(pts), dtype=bool)
+
+        # Below, we derive 1D arrays for the rst_rst function to take over after interpolating to the point coordinates
```

Comment:
do you mean "pts_pts"?

Reply:
Actually "rst_rst" but it's because BiasCorr methods can work on 1D arrays that match exactly in coordinates (which wouldn't be the case for "pts_pts").
I added a comment to clarify this behaviour